### PR TITLE
Add user role selection to workflow editor

### PR DESCRIPTION
**Pull request by Erikmitk**
This PR implements a basic user role selection in the workflow editor. Thereby, any task can be assigned to a user role to set permission for working on it.

screenshot 2018-10-05 at 11 55 31

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/services/data/UserGroupService.java [hunk 1]
```diff
@@ -354,7 +354,7 @@ private void addBasicUsersRelation(UserGroupDTO userGroupDTO, JsonObject jsonObj
         return convertListIdToDTO(getAllUserGroupIdsByClientIds(clientIdList), this);
     }
 
-    protected List<UserGroup> getAllUserGroupsByClientIds(List<Integer> clientIdList) {
+    public List<UserGroup> getAllUserGroupsByClientIds(List<Integer> clientIdList) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/UserGroupService.java]: Javadoc for public method?

File: Kitodo/src/main/webapp/WEB-INF/resources/js/modeler-custom.js [hunk 2]
```diff
@@ -9,12 +9,22 @@
...
+
+
+		var userRoles = $("#editForm\\:workflowTabView\\:userGroupId_input").children();
+
+		userRoles.each(function( index ) {
+			var role = { name: $(this).text(), value: $(this).val() };
+			availableUserRoles.push(role);
+		});
+	});
+});
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/WEB-INF/resources/js/modeler-custom.js]: Missing new line at end of file.

File: Kitodo/src/main/java/org/kitodo/services/data/UserGroupService.java (hunk 1 above)
Comment (Erikmitk) [reply Kitodo/src/main/java/org/kitodo/services/data/UserGroupService.java]: @Beacze Can you provide me with a JavaDoc comment? My PR is based on your branch that added this functionality.

Comment (Beacze) [reply Kitodo/src/main/java/org/kitodo/services/data/UserGroupService.java]: `/** Get all user roles by list of client ids
@param clientIds list of client ids

@return list of user groups for given client ids */`

**Pull request merged**